File: public/docs/sdks/go/lydian_iq.go
package lydian

import (
	"context"
	"encoding/json"
)

// LydianIQClient handles LyDian IQ API operations
type LydianIQClient struct {
	client *Client
}

// Signal represents a signal
type Signal struct {
	ID          string                 `json:"id"`
	Type        string                 `json:"type"`
	Source      string                 `json:"source"`
	Content     interface{}            `json:"content"`
	Timestamp   string                 `json:"timestamp"`
	Metadata    map[string]interface{} `json:"metadata,omitempty"`
	Processed   bool                   `json:"processed"`
	ProcessedAt *string                `json:"processed_at,omitempty"`
}

// KnowledgeEntity represents a knowledge entity
type KnowledgeEntity struct {
	ID            string                   `json:"id"`
	Type          string                   `json:"type"`
	Name          string                   `json:"name"`
	Properties    map[string]interface{}   `json:"properties"`
	Relationships []map[string]interface{} `json:"relationships,omitempty"`
	CreatedAt     string                   `json:"created_at,omitempty"`
	UpdatedAt     string                   `json:"updated_at,omitempty"`
}

// Insight represents an insight
type Insight struct {
	ID          string                   `json:"id"`
	Type        string                   `json:"type"`
	Title       string                   `json:"title"`
	Description string                   `json:"description"`
	Confidence  float64                  `json:"confidence"`
	Evidence    []map[string]interface{} `json:"evidence,omitempty"`
	Actionable  bool                     `json:"actionable"`
	Actions     []string                 `json:"actions,omitempty"`
	CreatedAt   string                   `json:"created_at,omitempty"`
}

// IngestSignal ingests a signal
func (lc *LydianIQClient) IngestSignal(ctx context.Context, signal Signal) (*Signal, error) {
	data, err := lc.client.Request(ctx, "POST", "/lydian-iq/signals", signal, nil)
	if err != nil {
		return nil, err
	}

	var result Signal
	if err := json.Unmarshal(data, &result); err != nil {
		return nil, err
	}

	return &result, nil
}

// CreateEntity creates a knowledge entity
func (lc *LydianIQClient) CreateEntity(ctx context.Context, entity KnowledgeEntity) (*KnowledgeEntity, error) {
	data, err := lc.client.Request(ctx, "POST", "/lydian-iq/knowledge/entities", entity, nil)
	if err != nil {
		return nil, err
	}

	var result KnowledgeEntity
	if err := json.Unmarshal(data, &result); err != nil {
		return nil, err
	}

	return &result, nil
}

// GenerateInsights generates insights from a query
func (lc *LydianIQClient) GenerateInsights(ctx context.Context, query string, context map[string]interface{}) ([]Insight, error) {
	body := map[string]interface{}{
		"query":   query,
		"context": context,
	}

	data, err := lc.client.Request(ctx, "POST", "/lydian-iq/insights/generate", body, nil)
	if err != nil {
		return nil, err
	}

	var results []Insight
	if err := json.Unmarshal(data, &results); err != nil {
		return nil, err
	}

	return results, nil
}

File: public/docs/sdks/typescript/src/lydian-iq.ts
import { LydianClient } from './client';
import { Signal, KnowledgeEntity, KnowledgeQuery, KnowledgeQueryResult, Insight, PaginatedResponse, PaginationParams } from './types';

/**
 * LyDian IQ API client
 */
export class LydianIQClient {
  constructor(private client: LydianClient) {}

  /**
   * Ingest signal into knowledge graph
   */
  async ingestSignal(data: Omit<Signal, 'id' | 'timestamp' | 'processed' | 'processedAt'>): Promise<Signal> {
    return this.client.request<Signal>('POST', '/lydian-iq/signals', { body: data });
  }

  /**
   * Batch ingest signals
   */
  async batchIngestSignals(signals: Array<Omit<Signal, 'id' | 'timestamp' | 'processed' | 'processedAt'>>): Promise<Signal[]> {
    return this.client.request<Signal[]>('POST', '/lydian-iq/signals/batch', { body: { signals } });
  }

  /**
   * Get signal by ID
   */
  async getSignal(signalId: string): Promise<Signal> {
    return this.client.request<Signal>('GET', `/lydian-iq/signals/${signalId}`);
  }

  /**
   * Query knowledge graph
   */
  async queryKnowledge(query: KnowledgeQuery): Promise<KnowledgeQueryResult> {
    return this.client.request<KnowledgeQueryResult>('POST', '/lydian-iq/knowledge/query', { body: query });
  }

  /**
   * Create knowledge entity
   */
  async createEntity(data: Omit<KnowledgeEntity, 'id' | 'createdAt' | 'updatedAt'>): Promise<KnowledgeEntity> {
    return this.client.request<KnowledgeEntity>('POST', '/lydian-iq/knowledge/entities', { body: data });
  }

  /**
   * Get entity by ID
   */
  async getEntity(entityId: string): Promise<KnowledgeEntity> {
    return this.client.request<KnowledgeEntity>('GET', `/lydian-iq/knowledge/entities/${entityId}`);
  }

  /**
   * Update entity
   */
  async updateEntity(entityId: string, data: Partial<KnowledgeEntity>): Promise<KnowledgeEntity> {
    return this.client.request<KnowledgeEntity>('PATCH', `/lydian-iq/knowledge/entities/${entityId}`, { body: data });
  }

  /**
   * Get insights
   */
  async getInsights(params?: PaginationParams & { type?: string; minConfidence?: number }): Promise<PaginatedResponse<Insight>> {
    return this.client.request<PaginatedResponse<Insight>>('GET', '/lydian-iq/insights', {
      query: params,
    });
  }

  /**
   * Get insight by ID
   */
  async getInsight(insightId: string): Promise<Insight> {
    return this.client.request<Insight>('GET', `/lydian-iq/insights/${insightId}`);
  }

  /**
   * Generate insights from query
   */
  async generateInsights(query: string, context?: Record<string, any>): Promise<Insight[]> {
    return this.client.request<Insight[]>('POST', '/lydian-iq/insights/generate', {
      body: { query, context },
    });
  }
}
